packages/genesys-spark-components/src/components/beta/gux-popover-beta/gux-popover.tsx

```diff
@@ -0,0 +1,321 @@
+        Object.assign(this.popupElement.style, {
+          left: `${x}px`,
+          top: `${y}px`
+        });
+
+        const side = placement.split('-')[0];
+
+        const staticSide = {
+          top: 'bottom',
+          right: 'left',
+          bottom: 'top',
+          left: 'right'
+        }[side];
+
+        const arrowRotation = {
+          top: 0,
+          right: 90,
+          bottom: 180,
+          left: -90
+        }[side];
+
+        // This is 13 because this makes the arrow look aligned when horizontal
+        // or 15 if vertical due to extra padding needed to show the shadow.
+        const arrowLen = side === 'left' || side === 'right' ? 15 : 13;
+
+        if (middlewareData.arrow) {
+          let x = middlewareData.arrow.x;
+          const y = middlewareData.arrow.y;
+
+          if (side === 'left' || side === 'right') {
+            x = x + 4;
+          }
+
+          this.popupElement.setAttribute('data-placement', placement);
+
+          Object.assign(this.arrowElement.style, {
+            left: x != null ? `${x}px` : '',
+            top: y != null ? `${y}px` : '',
+            right: '',
+            bottom: '',
+            [staticSide]: `${-arrowLen}px`,
+            transform: `rotate(${arrowRotation}deg)`
+          });
+        }
+      });
+    }
+  }
+
+  private dismiss(): void {
+    const dismissEvent = this.guxdismiss.emit();
+    if (!dismissEvent.defaultPrevented) {
+      this.isOpen = false;
+      this.popupElement.hidePopover();
+    }
+  }
+
+  private logForAttributeError(): void {
+    if (this.root.isConnected) {
+      console.error(
+        `gux-popover: invalid element supplied to 'for': "${this.for}"`
+      );
+    }
+  }
+
+  connectedCallback(): void {
+    this.updateForElement();
+    trackComponent(this.root, { variant: this.position });
+  }
+
+  componentDidLoad(): void {
+    if (!this.forElement) {
+      return;
+    }
+
+    this.popupElement.popover = 'manual';
+    this.forElement.popoverTargetElement = this.popupElement;
+    this.forElement.popoverTargetAction = 'toggle';
+
+    if (this.isOpen) {
+      this.popupElement.showPopover();
+      this.runUpdatePosition();
+    }
+  }
+
+  componentDidUpdate(): void {
+    if (this.isOpen) {
+      this.runUpdatePosition();
+    } else if (this.cleanupUpdatePosition) {
+      this.cleanupUpdatePosition();
+    }
+  }
+
+  disconnectedCallback(): void {
+    if (this.cleanupUpdatePosition) {
+      this.cleanupUpdatePosition();
+    }
+  }
+
+  private renderDismissButton(): JSX.Element {
+    if (this.displayDismissButton) {
+      return (
+        <gux-dismiss-button
+          onClick={this.dismiss.bind(this)}
+          position="inherit"
+        ></gux-dismiss-button>
+      ) as JSX.Element;
+    }
+  }
+
+  render(): JSX.Element {
+    return (
+      <popover
+        ref={(el: HTMLDivElement) => (this.popupElement = el)}
+        class={{
+          'gux-hidden': !this.isOpen,
```

Comment:
I think this styling can be removed as its probably done via , `showPopover` and `hidePopover` I think.

Reply:
Fixed

Reply:
Fixed

Reply:
So you're right that we don't need to control visibility anymore with css (I've removed the gux-hidden css for this component) but I'm having trouble checking if the popover is visible or not in this component's unit tests. Checking if `gux-hidden` is present or not makes it easy to test visibility for this component but let me know if you know of another way to check if the popover is visible or not in jest.